rpk: fix typos and Go bump

r-vasquez opened the pull request:
This PR fixes a couple of typos, fixes an issue where the `--all` flag was installed in all `cluster config` subcommands even if it wasn't used, and bumps Go dependencies.

## Backports Required

<!-- Checking at least one of the checkboxes is REQUIRED if this PR is not a backport. -->

- [ ] none - not a bug fix
- [ ] none - this is a backport
- [ ] none - issue does not exist in previous branches
- [ ] none - papercut/not impactful enough to backport
- [X] v25.2.x
- [ ] v25.1.x
- [ ] v24.3.x

## Release Notes

* none

r-vasquez committed:
rpk: fix typo in alter quotas and ACL example

r-vasquez committed:
rpk cluster config: install --all flag locally

It was previously installed at the root of the
'rpk cluster config' command, resulting in
appearing in all subcommands' help text even if
those didn't use it. This was confusing, so now we
just install it 'locally' in the commands that use
the flag.

r-vasquez committed:
rpk: bump go dependencies

rockwotj committed:
bazel: bump all golang deps

We want to keep all dependencies between the different go.mod files on
the same version, because in bazel we use a single dependency across all
modules. Docker is now resolving to a newer version, but
`github.com/moby/sys/user` was still on an old version, causing errors. I
would expect that `github.com/moby/moby` would have a dependency on the
newer version and that drag all the versions up but :shrug:

rockwotj committed:
bazel: bump rules_go

rockwotj reviewed (dismissed):
LGTM! Sorry for the Bazel hassles. Bazel treats all the go.mod files it manages as a single one, resolving dependencies and there were wierd things going on between the dependencies between the transform-sdk tests and rpk.

rockwotj committed:
bazel: remove src/transform-sdk/tests from bazel

These tests aren't used in CI with bazel, so for now we will just exclude them
it's not worth the hassle as to why the BUILD are not being generated
correctly.

rockwotj commented:
I ended up removing the unused transform sdk tests from the bazel build. They aren't used in CI this way and TBH it's not worth trying to figure out why the BUILD files were not being generated correctly

rockwotj committed:
bazel: bump buildifier version

vbotbuildovich commented:
<!--TEST RESULTS-->
<!--b82c8fb1-f4a2-427b-bfe3-4063258c3a9b-->
<!--title-->
## CI test results
<!--/title-->
<!--body-->
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/70412">build#70412</a></summary>


| test_class                | test_method                   | test_arguments   | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                                                                                                                                                  |
|:--------------------------|:------------------------------|:-----------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| EndToEndCloudTopicsTxTest | test_write                    | null             | integration | https://buildkite.com/redpanda/redpanda/builds/70412#0198862e-f077-4f9a-93b9-ff027aafc279 | FLAKY         | 20/21    | upstream reliability is '94.2008486562942'. current run reliability is '95.23809523809523'. drift is -1.03725 and the allowed drift is set to 50. The test should PASS  |
| FeaturesMultiNodeTest     | test_license_upload_and_query | null             | integration | https://buildkite.com/redpanda/redpanda/builds/70412#0198862e-f076-4fa7-af95-5fdf538be806 | FLAKY         | 20/21    | upstream reliability is '91.63636363636364'. current run reliability is '95.23809523809523'. drift is -3.60173 and the allowed drift is set to 50. The test should PASS |
| NodePreRestartProbeTest   | pre_restart_probe_test        | null             | integration | https://buildkite.com/redpanda/redpanda/builds/70412#01988630-e356-43dd-ac2f-19cfde94e364 | FLAKY         | 20/21    | upstream reliability is '99.71014492753623'. current run reliability is '95.23809523809523'. drift is 4.47205 and the allowed drift is set to 50. The test should PASS  |
</details>
<!--/body-->

vbotbuildovich commented:
/backport v25.2.x

vbotbuildovich commented:
Failed to create a backport PR to v25.2.x branch. I tried:

```
git remote add upstream https://github.com/redpanda-data/redpanda.git
git fetch --all
git checkout -b backport-pr-27147-v25.2.x-224 remotes/upstream/v25.2.x
git cherry-pick -x a284ba973f 7a28d335ef 17d48f61ac f59cbd9c30 7aa3d779ed 38e4126224 810999b8c2
```

[Workflow run logs](https://github.com/redpanda-data/redpanda/actions/runs/16816878283).
